R/fread.R

```diff
@@ -123,8 +124,12 @@ autostart=NA)
   }
   if (!is.null(colClasses) && is.atomic(colClasses)) {
     if (!is.character(colClasses)) stop("colClasses is not type list or character vector")
-    if (!length(colClasses)) stop("colClasses is character vector ok but has 0 length")
-    if (!is.null(names(colClasses))) {   # names are column names; convert to list approach
+    if (!length(colClasses)) {
+      colClasses=NULL;
+    } else if (identical(colClasses, "NULL")) {
```

Comment:
what's the use case for this? It's not mentioned in the manual...

Reply:
Not sure. Just saw it done in Hugh's branch and liked it. Maybe just to catch beginner errors?  There is a warning that it is taken to mean NULL.
I can see that it breaks the case-1-not-different rule.  So a 1-column file should have the column dropped (and null data.table returned) when colClasses="NULL"?  For consistency.  On the other hand beginners might well try "NULL".

Reply:
Maybe user has a loop through types,  and all columns should be coerced to that type. `for (class in c("integer","double","NULL")) fread(....,  colClasses=class)`

Reply:
Seems innocuous enough, just wondering if we should document it if there's a specific use case in mind... @HughParsonage any thoughts?